File: kernel/kernel-union/src/main/java/com/ycr/kernel/union/mvp/UnionActivity.java
package com.ycr.kernel.union.mvp;

import android.os.Bundle;

import com.ycr.kernel.mvp.MvpActivity;
import com.ycr.kernel.task.IGroup;
import com.ycr.kernel.task.TaskScheduler;
import com.ycr.kernel.union.UnionModuleKt;
import com.ycr.kernel.union.http.HttpHelper;
import com.ycr.kernel.union.mvp.view.IDefineView;
import com.ycr.kernel.union.mvp.view.ViewCreateHelper;
import com.ycr.kernel.util.KeyBoardUtils;


/**
 * Created by yuchengren on 2018/12/10.
 */
public abstract class UnionActivity extends MvpActivity implements IGroup,IDefineView{

	@Override
	protected void afterCreate(Bundle savedInstanceState) {
		super.afterCreate(savedInstanceState);
		ViewCreateHelper.createView(this, this, savedInstanceState);
	}
	@Override
	public String groupName() {
		return getClass().getName() + hashCode();
	}

	@Override
	protected void onDestroy() {
		String groupName = groupName();
		TaskScheduler.INSTANCE.cancelGroup(groupName);
		HttpHelper.cancelGroup(groupName);
		UnionModuleKt.getUnionLog().d("","cancel group-%s at onDestroy",groupName);
		KeyBoardUtils.fixInputMethodManagerLeak(this);
		super.onDestroy();
	}

	@Override
	protected void onPause() {
		super.onPause();
		TaskScheduler.INSTANCE.onPause(groupName());
	}

	@Override
	protected void onResume() {
		super.onResume();
		TaskScheduler.INSTANCE.onResume(groupName());
	}

}

File: kernel/kernel_union/src/main/java/com/ycr/kernel/union/mvp/UnionActivity.kt
package com.ycr.kernel.union.mvp

import android.os.Bundle

import com.ycr.kernel.mvp.MvpActivity
import com.ycr.kernel.task.IGroup
import com.ycr.kernel.task.TaskScheduler
import com.ycr.kernel.union.*
import com.ycr.kernel.union.http.HttpHelper
import com.ycr.kernel.union.mvp.view.IDefineView
import com.ycr.kernel.union.mvp.view.ViewCreateHelper
import com.ycr.kernel.util.KeyBoardUtils


/**
 * Created by yuchengren on 2018/12/10.
 */
abstract class UnionActivity : MvpActivity(), IGroup, IDefineView {

    override fun afterCreate(savedInstanceState: Bundle?) {
        super.afterCreate(savedInstanceState)
        createView(savedInstanceState)
    }

    open fun createView(savedInstanceState: Bundle?){
        ViewCreateHelper.createView(this, this, savedInstanceState)
    }

    override fun groupName(): String? {
        return javaClass.name + hashCode()
    }

    override fun onDestroy() {
        val groupName = groupName()
        TaskScheduler.cancelGroup(groupName)
        HttpHelper.cancelGroup(groupName!!)
        UnionLog.d("", "cancel group-%s at onDestroy", groupName)
        KeyBoardUtils.fixInputMethodManagerLeak(this)
        super.onDestroy()
    }

    override fun onPause() {
        super.onPause()
        TaskScheduler.onPause(groupName())
    }

    override fun onResume() {
        super.onResume()
        TaskScheduler.onResume(groupName())
    }

}
